diff --git a/xen/common/efi/boot.c b/xen/common/efi/boot.c
index 7bb4c643bf..f0800f47f8 100644
--- a/xen/common/efi/boot.c
+++ b/xen/common/efi/boot.c
@@ -157,6 +157,7 @@ static SIMPLE_TEXT_OUTPUT_INTERFACE *__initdata StdErr;
 
 static UINT32 __initdata mdesc_ver;
 static bool __initdata map_bs;
+static bool __initdata map_res = true;
 
 static struct file __initdata cfg;
 static struct file __initdata kernel;
@@ -1727,6 +1728,11 @@ void __init efi_init_memory(void)
                 if ( !map_bs )
                     continue;
                 break;
+
+            case EfiReservedMemoryType:
+                if ( !map_res )
+                    continue;
+                break;
             }
         }
 
