diff --git a/xen/common/efi/boot.c b/xen/common/efi/boot.c
index f0800f47f8..0fdd73e132 100644
--- a/xen/common/efi/boot.c
+++ b/xen/common/efi/boot.c
@@ -1473,6 +1473,18 @@ void EFIAPI __init noreturn efi_start(EFI_HANDLE ImageHandle,
         name.s = get_value(&cfg, section.s, "options");
         efi_arch_handle_cmdline(options, name.s);
 
+        name.s = get_value(&cfg, section.s, "mapbs");
+        if ( name.s )
+        {
+            map_bs = name.s[0] == '1';
+        }
+
+        name.s = get_value(&cfg, section.s, "noexitboot");
+        if ( name.s )
+        {
+            exit_boot_services = name.s[0] == '0';
+        }
+
         if ( !base_video )
         {
             name.cs = get_value(&cfg, section.s, "video");
