diff --git a/tools/libs/light/libxl_domain.c b/tools/libs/light/libxl_domain.c
index 25cde77835..f4bcb0b198 100644
--- a/tools/libs/light/libxl_domain.c
+++ b/tools/libs/light/libxl_domain.c
@@ -773,6 +773,7 @@ int libxl__domain_pvcontrol(libxl__egc *egc, libxl__xswait_state *pvcontrol,
     struct xs_permissions perms[] = {
         { .id = domid, .perms = XS_PERM_NONE },
     };
+    char *feature;
     int rc;
 
     rc = libxl__domain_pvcontrol_available(gc, domid);
@@ -791,6 +792,15 @@ int libxl__domain_pvcontrol(libxl__egc *egc, libxl__xswait_state *pvcontrol,
     if (!t)
         return ERROR_FAIL;
 
+    feature = libxl__xs_read(gc, t, GCSPRINTF("%s/control/feature-%s",
+                                              libxl__xs_get_dompath(gc, domid),
+                                              cmd));
+    if (!feature || strcmp(feature, "1")) {
+        LOGD(ERROR, domid, "PV control '%s' not supported by this domain", cmd);
+        xs_transaction_end(ctx->xsh, t, 1);
+        return ERROR_NOPARAVIRT;
+    }
+
     rc = libxl__xs_printf(gc, t, shutdown_path, "%s", cmd);
     if (rc) {
         xs_transaction_end(ctx->xsh, t, 1);
