diff --git a/tools/libs/light/libxl_dm.c b/tools/libs/light/libxl_dm.c
index dae9c27c23..854a328220 100644
--- a/tools/libs/light/libxl_dm.c
+++ b/tools/libs/light/libxl_dm.c
@@ -2537,6 +2537,9 @@ static void spawn_stub_launch_dm(libxl__egc *egc,
         while (*(serial++))
             num_console++;
     }
+    else
+        /* Qubes hack */
+        num_console = 1;
 
     console = libxl__calloc(gc, num_console, sizeof(libxl__device_console));
 
