diff --git a/tools/libs/light/libxl_create.c b/tools/libs/light/libxl_create.c
index 7968c0289e..7adb79a161 100644
--- a/tools/libs/light/libxl_create.c
+++ b/tools/libs/light/libxl_create.c
@@ -390,7 +390,7 @@ int libxl__domain_build_info_setdefault(libxl__gc *gc,
         libxl_defbool_setdefault(&b_info->u.hvm.hpet,               true);
         libxl_defbool_setdefault(&b_info->u.hvm.vpt_align,          true);
         libxl_defbool_setdefault(&b_info->u.hvm.usb,                false);
-        libxl_defbool_setdefault(&b_info->u.hvm.vkb_device,         true);
+        libxl_defbool_setdefault(&b_info->u.hvm.vkb_device,         false);
         libxl_defbool_setdefault(&b_info->u.hvm.xen_platform_pci,   true);
         libxl_defbool_setdefault(&b_info->u.hvm.pirq,               false);
 
