diff --git a/xen/common/grant_table.c b/xen/common/grant_table.c
index 6c77867f8c..a525088c0f 100644
--- a/xen/common/grant_table.c
+++ b/xen/common/grant_table.c
@@ -179,7 +179,7 @@ static int cf_check parse_gnttab_max_maptrack_frames(const char *arg)
 }
 
 #ifndef GNTTAB_MAX_VERSION
-#define GNTTAB_MAX_VERSION 2
+#define GNTTAB_MAX_VERSION 1
 #endif
 
 unsigned int __read_mostly opt_gnttab_max_version = GNTTAB_MAX_VERSION;
