diff --git a/Config.mk b/Config.mk
index 4bbb366119..c8e8f1de77 100644
--- a/Config.mk
+++ b/Config.mk
@@ -255,3 +255,5 @@ QEMU_UPSTREAM_LOC ?= $(call or,$(wildcard $(QEMU_UPSTREAM_INTREE)),\
                                $(QEMU_UPSTREAM_URL))
 
 CONFIG_TESTS       ?= y
+
+DATE_EPOCH_OPTS := $(if $(SOURCE_DATE_EPOCH),-u -d "@$(SOURCE_DATE_EPOCH)")
\ No newline at end of file
diff --git a/INSTALL b/INSTALL
index 1ea71430ed..ab471948f9 100644
--- a/INSTALL
+++ b/INSTALL
@@ -235,6 +235,12 @@ XEN_BUILD_TIME=hh:mm:ss
 SMBIOS_REL_DATE=mm/dd/yyyy
 VGABIOS_REL_DATE="dd Mon yyyy"
 
+It is sufficient to define only SOURCE_DATE_EPOCH instead of the above
+variables for having reproducible builds. By default, they use
+SOURCE_DATE_EPOCH if defined. For specifications of SOURCE_DATE_EPOCH,
+we refer to https://reproducible-builds.org/docs/source-date-epoch.
+SOURCE_DATE_EPOCH=
+
 This variable can be used to point to a different git binary to be used.
 GIT=
 
diff --git a/tools/firmware/hvmloader/Makefile b/tools/firmware/hvmloader/Makefile
index cc5dc00498..bd18d8193a 100644
--- a/tools/firmware/hvmloader/Makefile
+++ b/tools/firmware/hvmloader/Makefile
@@ -23,7 +23,7 @@ include $(XEN_ROOT)/tools/firmware/Rules.mk
 ld-option = $(shell if $(LD) -v $(1) >/dev/null 2>&1; then echo y; else echo n; fi)
 
 # SMBIOS spec requires format mm/dd/yyyy
-SMBIOS_REL_DATE ?= $(call date,"+%m/%d/%Y")
+SMBIOS_REL_DATE ?= $(shell date $(DATE_EPOCH_OPTS) +%m%d%Y)
 
 CFLAGS += $(CFLAGS_xeninclude) -fno-pic -mregparm=3
 
diff --git a/tools/firmware/vgabios/Makefile b/tools/firmware/vgabios/Makefile
index db5a2624be..35c30df51d 100644
--- a/tools/firmware/vgabios/Makefile
+++ b/tools/firmware/vgabios/Makefile
@@ -8,7 +8,7 @@ BCC = bcc
 AS86 = as86
 
 RELEASE = `pwd | sed "s-.*/--"`
-VGABIOS_REL_DATE ?= $(call date,"+%d %b %Y")
+VGABIOS_REL_DATE ?= $(shell date $(DATE_EPOCH_OPTS) +%d %b %Y)
 RELVERS = `pwd | sed "s-.*/--" | sed "s/vgabios//" | sed "s/-//"`
 
 VGABIOS_DATE = "-DVGABIOS_DATE=\"$(VGABIOS_REL_DATE)\""
diff --git a/xen/Makefile b/xen/Makefile
index 074052d0b0..78fac0b3f4 100644
--- a/xen/Makefile
+++ b/xen/Makefile
@@ -247,10 +247,10 @@ SRCARCH := $(shell echo $(ARCH) | \
 export ARCH SRCARCH
 
 ifeq ($(origin XEN_BUILD_DATE), undefined)
-export XEN_BUILD_DATE	:= $(call date)
+export XEN_BUILD_DATE	:= $(shell LC_ALL=C date $(DATE_EPOCH_OPTS))
 endif
 ifeq ($(origin XEN_BUILD_TIME), undefined)
-export XEN_BUILD_TIME	:= $(call date,"+%T")
+export XEN_BUILD_DATE	:= $(shell LC_ALL=C date $(DATE_EPOCH_OPTS) +%T)
 endif
 
 export CONFIG_SHELL := $(SHELL)
