diff --git a/INSTALL b/INSTALL
index ab471948f9..8d4f6db277 100644
--- a/INSTALL
+++ b/INSTALL
@@ -234,6 +234,7 @@ XEN_BUILD_DATE=<output of date(1)>
 XEN_BUILD_TIME=hh:mm:ss
 SMBIOS_REL_DATE=mm/dd/yyyy
 VGABIOS_REL_DATE="dd Mon yyyy"
+PANDOC_REL_DATE="+%Y-%m-%d"
 
 It is sufficient to define only SOURCE_DATE_EPOCH instead of the above
 variables for having reproducible builds. By default, they use
diff --git a/docs/Makefile b/docs/Makefile
index 37776d303c..0e25b7ac01 100644
--- a/docs/Makefile
+++ b/docs/Makefile
@@ -3,7 +3,7 @@ include $(XEN_ROOT)/Config.mk
 -include $(XEN_ROOT)/config/Docs.mk
 
 VERSION		:= $(shell $(MAKE) -C $(XEN_ROOT)/xen --no-print-directory xenversion)
-DATE		:= $(call date,"+%Y-%m-%d")
+PANDOC_REL_DATE		:= $(shell date $(DATE_EPOCH_OPTS) +%Y-%m-%d)
 
 DOC_ARCHES      := arm ppc riscv x86_32 x86_64
 MAN_SECTIONS    := 1 5 7 8
@@ -98,7 +98,7 @@ man$(1)/%.$(1): man/%.$(1).pandoc Makefile
 ifneq ($(PANDOC),)
 	@$(INSTALL_DIR) $$(@D)
 	$(PANDOC) --standalone -V title=$$* -V section=$(1) \
-		      -V date="$(DATE)" -V footer="$(VERSION)" \
+		      -V date="$(PANDOC_REL_DATE)" -V footer="$(VERSION)" \
 			  -V header=Xen $$< -t man --output $$@
 else
 	@echo "pandoc not installed; skipping $$@"
